src/algebra/ring/pi.lean

```diff
@@ -109,11 +109,9 @@ lemma non_unital_ring_hom_injective {γ : Type w} [nonempty I]
 @[simps]
 protected def ring_hom {γ : Type w} [Π i, non_assoc_semiring (f i)] [non_assoc_semiring γ]
   (g : Π i, γ →+* f i) : γ →+* Π i, f i :=
-{ to_fun := λ x b, g b x,
-  map_add' := λ x y, funext $ λ z, (g z).map_add x y,
-  map_mul' := λ x y, funext $ λ z, (g z).map_mul x y,
+{ map_mul' := λ x y, funext $ λ z, (g z).map_mul x y,
   map_one' := funext $ λ z, (g z).map_one,
-  map_zero' := funext $ λ z, (g z).map_zero }
+  .. pi.add_monoid_hom (λ i, (g i).to_add_monoid_hom) }
```

Comment:
`.. pi.monoid_hom (λ i, (g i).to_monoid_hom)` should work for the other two fields.

Reply:
Yeah, I tried that but somehow it does not work. I don't know why
```
type mismatch at application
  monoid_hom (λ (i : I), (g i).to_monoid_hom)
term
  λ (i : I), (g i).to_monoid_hom
has type
  Π (i : I), γ →* f i : Type (max u v w)
but is expected to have type
  Π (i : I), ?m_1 →* ?m_2 i : Type (max u ? ?)
```

Reply:
I just pushed a fix; your mistake was using `monoid γ` where you actually only need `mul_one_class γ`.

Reply:
I see.. I guess the structure hierarchy is still not very clear too me...